src/DynamoCore/Engine/EngineController.cs

```diff
@@ -522,7 +522,8 @@ private void OnLibraryLoaded()
         /// </summary>
         private void LibraryLoaded(object sender, LibraryServices.LibraryLoadedEventArgs e)
         {
-            OnLibraryLoaded();
+            if(e.LibraryPaths.Any())
```

Comment:
@aparajit-pratap so this will not have any paths if a package only containing custom nodes is loaded? What about a package containing nodes and zt assemblies?

Are there are tests that should be added here to check this event is not raised?

Reply:
A package containing custom nodes and ZT assemblies will have relevant paths as expected, in which case we would need to reload all libraries including the new ones in that package. 

I can try and come up with some tests.